### PR TITLE
Implement torch.kernel_call capture.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Had to stop short of modifying the function return signature because of a missing C-API upstream.
* Committing here is good enough for a test and will resolve the various TODOs about upstream APIs next.

### CODE REVIEW

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 1]
```diff
@@ -62,23 +106,170 @@ std::shared_ptr<AcapController> AcapController::getCurrent() {
...
+
+  // Invoke the original kernel.
+  redispatch(opHandle, stack);
+
+  // Map returns to results.
+  size_t returnCount = schema.returns().size();
+  assert(stack->size() >= returnCount && "stack too short");
+  llvm::SmallVector<MlirType, 4> resultTypes;
+  llvm::SmallVector<std::pair<size_t, at::Tensor>, 4> resultIndexToTensorMap;
+  for (auto returnIt = stack->end() - returnCount; returnIt != stack->end();
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:190]: Is the first argument top of stack or not? This code suggests it is not, and the torch.kernel_call docs need to be udpated.

File: frontends/pytorch/csrc/c10_dispatch/func_builder.h [hunk 2]
```diff
@@ -16,6 +16,33 @@
...
+  OperationStateHolder(const OperationStateHolder &) = delete;
+  OperationStateHolder(OperationStateHolder &&other) = delete;
+  ~OperationStateHolder() {
+    if (owned) {
+      // TODO: Upstream a mlirOperationStateDestroy() function.
+      mlirOperationDestroy(createOperation());
+    }
+  }
+
+  MlirOperationState state;
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/func_builder.h:34]: weird ordering for this member. move after createOperation?

**Review by silvasean — APPROVED** [review]
This is looking so great!

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp (hunk 1 above)
Comment (stellaraccident) [reply frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:190]: Seems not. I had it reversed and that turned out to not be right.

**Pull request merged**